src/pages/nft/class/_classId/index.vue

```diff
@@ -888,6 +884,35 @@ export default {
         };
         logPurchaseFlowEvent(this, 'add_to_cart', purchaseEventParams);
         logPurchaseFlowEvent(this, 'begin_checkout', purchaseEventParams);
+        if (edition.price === 0) {
+          this.$router.push(
+            this.localeLocation({
+              name: 'nft-claim',
+              query: {
+                class_id: this.classId,
+                type: 'nft_book',
+                free: true,
+                price_index: edition.index,
+                from: 'liker_land_waived',
+              },
+            })
+          );
+          return;
+        }
+        if (edition.price > 0 && this.nftPriceInLIKE > 0) {
+          await this.initIfNecessary();
+          if (this.hasConnectedWallet) {
+            logPurchaseFlowEvent(
+              this,
+              'add_shipping_info',
+              purchaseEventParams
+            );
+            this.fetchUserCollectedCount();
+            this.walletFetchLIKEBalance();
+          }
+          this.uiToggleCollectModal({ classId: this.classId });
+          return;
+        }
```

Comment:
use `else if` instead of return here might be easier to maintain, since a long block of early returns is harder to read